config-controller/internal/controller/policy_controller.go

```diff
@@ -69,6 +68,29 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, errors.Wrap(err, "Failed to refresh")
 	}
 
+	desiredState := policyCR.Spec.ToProtobuf()
+	desiredState.Id = policyCR.Status.PolicyId
+
+	existingPolicy, exists, err := r.PolicyClient.GetPolicy(ctx, desiredState.GetName())
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "Failed to fetch policy")
+	}
+
+	if exists && existingPolicy.IsDefault {
+		retErr := errors.New(fmt.Sprintf("Failed to reconcile: existing default policy with the same name '%s' exists", desiredState.GetName()))
+		policyCR.Status.Accepted = false
+		policyCR.Status.Message = retErr.Error()
```

Comment:
We want to set these fields, but we need to persist them.

```suggestion
		policyCR.Status.Message = retErr.Error()
		if err := r.K8sClient.Status().Update(ctx, policyCR); err != nil {
		    return ctrl.Result{}, errors.Wrap(err, fmt.Sprintf("Failed to set status on policy '%s'", desiredState.GetName()))
	        }
```

Reply:
Gotta save the status still :smile: